scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# USAGE: standard-header-new_guards.pl filename.h >filename.h.new
+#
+# source-maintenance plungin to standardise the header guards
+#
+# Behaviour:
+# - files whose name does not end in .h are passed verbatim
+# - anything before the first #ifdef SOMETHING_ENDING_IN_H is copied verbatim
+# - the existing header guard is replaced with SQUID_RELATIVE_PATH_OF_FILE_H
+# - anything until the last #endif clause in the file is copied verbatim
+# - the last #endif clause is replaced with the closing header guard and an empty line
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+my $new_guard = $filename;
+$new_guard =~ s/[\/\-\.]/_/g;
+$new_guard = "SQUID_".uc($new_guard);
```

Comment:
Please declare this and other variables as late as possible. Doing so makes code more readable, clarifies its intent, reduces chances of leaving unused variables behind, etc.

Reply:
This already is as late as possible; both variables are set in the loop
immediately below, and may not be scoped in that loop because they are
referenced outside the loop after it finishes

Reply:
done (old stale comment had some skew in line numbers, sorry for that)